### PR TITLE
Added support for ZTD feature

### PR DESCRIPTION
**Pull request by skg-net**
Signed-off-by: Senthil Kumar Ganesan <skg.net.dev@gmail.com>

### CODE REVIEW

File: rootconf/etc/dhclient.conf [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+request subnet-mask, broadcast-address, time-offset, routers,
+        domain-name, domain-name-servers, domain-search, host-name,
+        dhcp6.name-servers, dhcp6.domain-search, dhcp6.fqdn, dhcp6.sntp-servers,
+        netbios-name-servers, netbios-scope, interface-mtu,
+        rfc3442-classless-static-routes, ztd-provision-url, ntp-servers;
+
+#send dhcp-client-identifier 1:0:a0:24:ab:fb:9c;
+#send dhcp-lease-time 3600;
+#supersede domain-name "fugue.com home.vix.com";
```
Comment (theucke) [inline rootconf/etc/dhclient.conf]: What are these about?

File: VERSION [hunk 2]
```diff
@@ -1 +1 @@
-1.0.8
+1.0.9
```
Comment (heucke) [inline VERSION]: Since you're adding a feature, bump the minor version number instead. `1.1.0`

File: rootconf/etc/ztd [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+   echo "ZTD Provision URL:  $new_ztd_provision_url"
+
+   #download the provisioning url
+   curl -o $ZTD_PROV_SCRIPT $new_ztd_provision_url
+
+   if [ $? -eq 0 ]; then
+         #execute the provisioning url
+         chmod +x $ZTD_PROV_SCRIPT
+         $ZTD_PROV_SCRIPT
+         if [ $? -eq 0 ]; then
```
Comment (heucke) [inline rootconf/etc/ztd]: You can check the exit code directly like this:

```bash
if $ZTD_PROV_SCRIPT; then
```

> https://github.com/koalaman/shellcheck/wiki/SC2181

File: rootconf/etc/ztd [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+ZTD_FILE="/etc/opx/ztd/ztd"
+ZTD_PROV_SCRIPT="/etc/opx/ztd/ztd_prov_script"
+
+#check if it a new assignment, ztd enabled and ztd option recieved
+if [ "$reason" = "BOUND" ] && [ -f $ZTD_FILE ] && [ -n "$new_ztd_provision_url" ]; then
+   echo "ZTD is enabled, request recieved on interface: $interface"
+   echo "ZTD Provision URL:  $new_ztd_provision_url"
+
+   #download the provisioning url
+   curl -o $ZTD_PROV_SCRIPT $new_ztd_provision_url
```
Comment (heucke) [inline rootconf/etc/ztd]: Double quote the variables. `$new_ztd_provision_url` is an external variable.

> https://github.com/koalaman/shellcheck/wiki/SC2086

File: rootconf/etc/ztd [hunk 5]
```diff
@@ -0,0 +1,28 @@
+#!/bin/sh
```
Comment (heucke) [inline rootconf/etc/ztd:1]: This single file has 3-space indentation, 4-space indentation, and 6-space indentation.

**Review by heucke — CHANGES_REQUESTED** [review]
Update the version, at least.

**Comment by skg-net** [comment]
Pull-request updated, HEAD is now da1a89f04eb01481f292f3e59f1a0be3ce86f6b2

**Comment by skg-net** [comment]
@theucke addressed the review comments, please check.

**Review by heucke — APPROVED** [review]

**Pull request merged**